contracts/SpokePool.sol

```diff
@@ -546,6 +566,72 @@ abstract contract SpokePool is
+     * The equivalent of this token on the relayer's repayment chain of choice will be sent as a refund. If this is equal
+     * to the wrapped native token, the caller can optionally pass in native token as msg.value, provided msg.value = inputTokenAmount.
+     * @param outputToken The token that the relayer will send to the recipient on the destination chain. Must be an ERC20.
+     * @param inputAmount The amount of input tokens pulled from the caller's account and locked into this contract. This
+     * amount will be sent to the relayer as a refund following an optimistic challenge window in the HubPool, less a system fee.
+     * @param outputAmount The amount of output tokens that the relayer will send to the recipient on the destination.
+     * @param destinationChainId The destination chain identifier. Must be enabled along with the input token as a valid
+     * deposit route from this spoke pool or this transaction will revert.
+     * @param exclusiveRelayer The relayer exclusively allowed to fill this deposit before the exclusivity deadline.
+     * @param quoteTimestamp The HubPool timestamp that determines the system fee paid by the depositor. This must be set
+     * between [currentTime - depositQuoteTimeBuffer, currentTime] where currentTime is block.timestamp on this chain.
+     * @param fillDeadline The deadline for the relayer to fill the deposit. After this destination chain timestamp, the fill will
+     * revert on the destination chain. Must be set between [currentTime, currentTime + fillDeadlineBuffer] where currentTime
+     * is block.timestamp on this chain.
+     * @param exclusivityPeriod Added to the current time to set the exclusive relayer deadline. After this timestamp,
+     * anyone can fill the deposit.
+     * @param message The message to send to the recipient on the destination chain if the recipient is a contract. If the
+     * message is not empty, the recipient contract must implement `handleV3AcrossMessage()` or the fill will revert.
+     */
+    function depositV3(
+        address depositor,
+        address recipient,
+        address inputToken,
+        address outputToken,
+        uint256 inputAmount,
+        uint256 outputAmount,
+        uint256 destinationChainId,
+        address exclusiveRelayer,
+        uint32 quoteTimestamp,
+        uint32 fillDeadline,
+        uint32 exclusivityPeriod,
+        bytes calldata message
+    ) public payable {
+        depositV3(
+            depositor.toBytes32(),
+            recipient.toBytes32(),
+            inputToken.toBytes32(),
+            outputToken.toBytes32(),
+            inputAmount,
+            outputAmount,
+            destinationChainId,
+            exclusiveRelayer.toBytes32(),
+            quoteTimestamp,
+            fillDeadline,
+            exclusivityPeriod,
+            message
+        );
+    }
```

Comment:
I still think we may want to consider just forwarding the raw msg.data to the other function via a self delegatecall. I think this would be cheaper and less bytecode (although I could be wrong).

Reply:
I'll look into this and report back.

Reply:
I played with this for a fiew hours and think that the msg.data + self delegate call is overly complex and not worth it. I did something like this:
```
fallback() external payable {
        // Extract the function selector
        bytes4 selector;
        assembly {
            selector := calldataload(0)
        }

        // Check if selector matches `depositV3(address, address, address, address, uint256, uint256, uint256, address, uint32, uint32, uint32, bytes)`
        if (
            selector ==
            bytes4(
                keccak256(
                    "depositV3(address,address,address,address,uint256,uint256,uint256,address,uint32,uint32,uint32,bytes)"
                )
            )
        ) {
            // Extract each address parameter and cast to bytes32
            bytes32 depositor;
            bytes32 recipient;
            bytes32 inputToken;
            bytes32 outputToken;
            bytes32 exclusiveRelayer;

            // Using assembly for efficient calldata extraction and casting
            // Using assembly for efficient calldata extraction and conversion
            assembly {
                depositor := shl(96, calldataload(4)) // address -> bytes32
                recipient := shl(96, calldataload(36)) // address -> bytes32
                inputToken := shl(96, calldataload(68)) // address -> bytes32
                outputToken := shl(96, calldataload(100)) // address -> bytes32
                exclusiveRelayer := shl(96, calldataload(228)) // address -> bytes32
            }

            // Load other uint256 and uint32 values from calldata directly
            uint256 inputAmount;
            uint256 outputAmount;
            uint256 destinationChainId;
            uint32 quoteTimestamp;
            uint32 fillDeadline;
            uint32 exclusivityPeriod;
            bytes memory message;

            // Directly load the rest of the parameters from calldata
            assembly {
                inputAmount := calldataload(132)
                outputAmount := calldataload(164)
                destinationChainId := calldataload(196)
                quoteTimestamp := calldataload(260)
                fillDeadline := calldataload(292)
                exclusivityPeriod := calldataload(324)
            }

            // uint256 messageStart = 356; // Correct starting position for message
            // uint256 messageLength = msg.data.length > messageStart ? msg.data.length - messageStart : 0;

            // Copy `message` data directly
            // message = new bytes(messageLength);
            // if (messageLength > 0) {
            //     for (uint i = 0; i < messageLength; i++) {
            //         message[i] = msg.data[messageStart + i];
            //     }
            // }
            message = new bytes(0);

            // Forward the call to `depositV3(bytes32, bytes32, bytes32, bytes32, uint256, uint256, uint256, bytes32, uint32, uint32, uint32, bytes)
            (bool success, ) = address(this).delegatecall(
                abi.encodeWithSelector(
                    this.depositV3.selector,
                    depositor,
                    recipient,
                    inputToken,
                    outputToken,
                    inputAmount,
                    outputAmount,
                    destinationChainId,
                    exclusiveRelayer,
                    quoteTimestamp,
                    fillDeadline,
                    exclusivityPeriod,
                    message
                )
            );
            require(success, "Delegatecall to depositV3 failed");
        } else {
            // Handle unexpected calls
            revert("Function selector not supported");
        }
    }
```
(note the `message` decoding is broken so I hard coded it to bytes 0).

this works. but:
1.  it's more expensive than the wrapped version (marginally, but still more).
2. it was super irritating to implement, 
3. it makes our ABIs a massive pain to work with (we need a custom interface in the build process to append this to the ABI as we no longer have public functions exposing them

is this the kind of thing you were envisioning? I think if this is the pattern we need to go for then I'm a strong proponent of the much easier version that the PR has rn.